### ISSUE TITLE
rename `Alist` to `SnocList`, `RevList`, or other well-known name

### ISSUE THREAD

**Issue by leque** [gfngfn/SATySFi#390]
I sometimes see people (including me) are puzzled[^1] by the name of `src/alist.ml` (accumulated list? not associative list!). How about changing its name to something more well-known?

Candidates might be:
- SnocList
- RevList
- ...

[^1]: (in Japanaese) https://twitter.com/elpin1al/status/1244498976771944450

**Comment by gfngfn** [comment]
Yes, by `Alist` I meant “accumulated lists.” I also understand the name is a bit confusing.

The module implements the following requirement:

- An `'a t`-typed value is essentially an ordered list of `'a`-typed values.
- Appending an element to the end can be done in $O(1)$.
- Conversion of `'a t` into `'a list` can be done in $O(n)$ where $n$ is the number of elements.

Renaming this module seems to be a good idea for me. Thank you for the suggestion.

However, I’m not willing to rename it to `RevList`; it’s true that the requirement above is implemented by reversed singly linked lists, but that is an implementation detail, not an inevitable design. I prefer something like `AccList`.

**Comment by leque** [comment]
I prefer `SnocList`; as the name suggesting, it is an ordered list with 𝑂(1) append-last operation (snoc). It is a FP-jargon, but it is easy to know its meaning by googling.

`AccList` is also OK to me.

**Comment by leque** [comment]
(Or, use `Queue.t`, `Queue.add` , and `Queue.to_seq |> List.of_seq`. It is mutable, though.)
